### DISCUSSION TITLE
Provide connection details programmatically

### DISCUSSION THREAD

**Discussion by jpaju** [jorgerojas26/lazysql#90] [Ideas]
Hi there 👋🏼

First off, I want to say thanks for building lazysql — it’s the most feature rich SQL TUI in my opinion.

One thing that I've been struggling with, though, is the process of connecting to databases using temporary credentials. At work, we can connect to DBs only using temporary credentials, which we get a CLI command, which means they change frequently. Right now, I have to start lazysql and then manually input the connection details every time.

I think it would be awesome if lazysql could support providing connection details directly via command-line arguments (something like `--host`, `--username`, and `--password`) or through environment variables (e.g., `LAZYSQL_HOST`, `LAZYSQL_USER`, and `LAZYSQL_PASSWORD`). This would streamline the process a lot and make it easier to work with temporary credentials.

What do you think about this sort of feature? Is this something that has been discussed or thought of before?

**Comment by jorgerojas26** [comment]
Hi, thank you very much. :) 

I think this would be awesome, i'll definitely think about implementing this. Right now i'm very very busy (i wish i wasn't) but i'd like this to be in lazysql for sure.

Thanks for suggesting.

**Reply by jpaju** [reply]
Thanks for the quick reply! Awesome that you're considering this 🙏🏼 One thing I forgot to mention was that providing the password as command line argument may not be a good idea since it could be found from shell history. But I'm sure this could be figured out if it was to be implemented!
